django/utils/functional.py

```diff
@@ -75,6 +75,10 @@ def __reduce__(self):
                 (func, self.__args, self.__kw) + resultclasses
             )
 
+        def __repr__(self):
+            """ Make sure the lazy object's display is useful when debugging """
```

Comment:
Is the method needed on Python 3? (i.e. does the test pass there without it?) If so, we could conditionally add this method on Python 2 by putting it in a `if six.PY2:` branch so that we remove it when dropping Python 2 support.

The docstring isn't really needed and I'd sort the method above `__str__`.